src/_path_wrapper.cpp

```diff
@@ -727,6 +727,7 @@ static PyObject *Py_convert_to_string(PyObject *self, PyObject *args, PyObject *
             return NULL;
         }
         codes[i] = PyBytes_AsString(item);
+        Py_DECREF(item);
```

Comment:
Just change the PyArg_ParseTuple call to use a (yyyyy) format for parsing the codes argument instead?

Reply:
I am 50/50 on that.  It lets python do more of the work, but is also a bigger change.

Reply:
I think all of our callers pass the right length lists, but as I'm targeting 3.0.3 with this, I'm not sure if it's quite as safe to be backporting that version.

Reply:
But we already error out a few lines above if the length is not 5.
Anyways I don't want to insist too much on this, if you don't want to make the bigger change that's fine with me too.